specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/stable/2024-06-30/cloudhsm.json

```diff
@@ -0,0 +1,1470 @@
+            "$ref": "./examples/CloudHsmCluster_RequestOrValidate_Restore_MaximumSet_Gen.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.HardwareSecurityModules/cloudHsmClusters/{cloudHsmClusterName}/restore/{jobId}/pending": {
+      "get": {
+        "tags": [
+          "Returns Restore Operation Status"
+        ],
+        "operationId": "CloudHsmClustersRestoreStatus_Get",
+        "description": "Gets the restore operation status of the specified Cloud HSM Cluster",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/cloudHsmClusterName"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/backupRestoreJobId"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Cloud HSM Cluster Backup Operation status returned",
+            "schema": {
+              "$ref": "#/definitions/RestoreResult"
+            },
+            "headers": {
+              "x-ms-request-id": {
+                "description": "A unique ID for the current operation, service generated. All the resource providers must return this value in the response headers to facilitate debugging.",
+                "type": "string"
+              }
+            }
+          },
+          "default": {
+            "description": "The error response describing why the operation failed",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v6/types.json#/definitions/ErrorResponse"
+            },
+            "headers": {
+              "x-ms-request-id": {
+                "description": "A unique ID for the current operation, service generated. All the resource providers must return this value in the response headers to facilitate debugging.",
+                "type": "string"
+              }
+            }
+          }
+        },
+        "x-ms-examples": {
+          "CloudHsmCluster_Get_Restore_Status_MaximumSet_Gen": {
+            "$ref": "./examples/CloudHsmCluster_Restore_Pending_MaximumSet_Gen.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "CloudHsmProperties": {
+      "type": "object",
+      "description": "The Cloud HSM Properties",
+      "properties": {
+        "fqdn": {
+          "type": "string",
+          "description": "FQDN of the Cloud HSM"
+        },
+        "state": {
+          "type": "string",
+          "description": "The Cloud HSM State"
```

Comment:
>State [](http://example.com/codeflow?start=40&length=5)

How would you use it?
Is this supposed to be seen by consumers as an enumeration, or is it a human readable / localized thing?

Reply:
This is a value that we receive from another service, cannot specify the enum properties here as it does not depend on our service.

Reply:
So.... for the sake of anybody who wants to consume that information, how do they know what to expect? Are you going to link them to a doc for that other service?
And are they randomly going to start getting *different* responses, just because you internally updated your api-version that you use to call the other service on the implementation backend?

Reply:
That sounds like a step in the right direction, but I think it still raises some worries about how once you've documented a set of statuses you're going to be able to version this API without breaking customer's expectations, that they may have baked into code.